src/sim/trace-extract/google_trace_simulator.cc

```diff
@@ -135,19 +260,32 @@ GoogleTraceSimulator::~GoogleTraceSimulator() {
 }
 
 void GoogleTraceSimulator::Run() {
-  FLAGS_debug_flow_graph = true;
   FLAGS_add_root_task_to_graph = false;
+  FLAGS_flow_scheduling_strict = true;
+
+  const uint32_t MAX_VALUE = UINT32_MAX;
+  proportion_ = (FLAGS_percentage / (double)100) * MAX_VALUE;
+  VLOG(2) << "Retaining events with hash < " << proportion_;
```

Comment:
The code uses Spooky::Hash32, which returns a uint32_t, so this is fine. However, looking at the Spooky source code, the uint32_t is obtained by casting a uint64_t, so we may as well move to Hash64 without drop in performance.